### PR TITLE
Add group reports for flake-group check

### PR DESCRIPTION
**Pull request by ysndr**
flake-info will populate an environment variable with a report for all failed flakes of a group if applicable.
This should help detecting faulty flakes more easily

- adds a `--report` flag to `flake-info group` which will create a report of failed flakes if some flakes of a group fail to evaluate.
- if all flakes fail the process will throw an error as it is more likely that a major issue occured

## Example

image

### CODE REVIEW

**Comment by ncfavier** [comment]
Sadly that is not how environment variables [work](https://mywiki.wooledge.org/Environment).

We should probably rather make that report part of the output JSON when `--json` is given.

**Comment by ysndr** [comment]
Eh yes 🙈 
It was more of a quick idea I ran, since i've seen this used before and didnt think too much about it.
You're right though.

**Comment by ysndr** [comment]
@ncfavier what do you think?

**Comment by ysndr** [comment]
Well here we do process the output (creating annotations)
@erikarvstedt advised not to print errors on stdout and stderr is already used for logging which is the reason for this here.

I could print the report to stdout which would require `--json` and `--report` to be mutually exclusive

**Comment by ysndr** [comment]
It has greater visibility since the error is not mixed into a wall of other logging output

File: .github/workflows/check-flake-files.yml [hunk 1]
```diff
@@ -34,7 +33,22 @@ jobs:
...
         do
-          ./result/bin/flake-info group "$flake_group" "$(basename "$flake_group" .toml)"
+          echo "::group::Group \"$(basename $flake_group)\""
+        
+         ./result/bin/flake-info group "$flake_group" "$(basename "$flake_group" .toml)" --report
+
+          if [[ -f "./report.txt" ]]
+          then
+            had_error=1
+            echo "::error file=$flake_group::$(cat ./report.txt)"
```
Comment (ncfavier) [inline .github/workflows/check-flake-files.yml:46]:
```suggestion
            echo "::error file=$flake_group::$(< ./report.txt)"
```

File: .github/workflows/check-flake-files.yml [hunk 2]
```diff
@@ -34,7 +33,22 @@ jobs:
...
+          fi
+
+          echo ::endgroup::
+
+          exit $had_error
```
Comment (ncfavier) [inline .github/workflows/check-flake-files.yml]: I think you want to `exit` outside the loop, so that we get all errors?

**Review by ncfavier — APPROVED** [review]

File: .github/workflows/check-flake-files.yml [hunk 3]
```diff
@@ -34,7 +33,29 @@ jobs:
...
+          then
+            had_error=1
+
+            # sic.:
+            # Workaround for multi line output
+            echo "REPORT<<EOF" >> $GITHUB_ENV
+            echo "$(< ./report.txt)" >> $GITHUB_ENV
+            echo "EOF" >> $GITHUB_ENV
+
+            echo "::error file=$flake_group::${ env.REPORT }"
```
Comment (ncfavier) [inline .github/workflows/check-flake-files.yml:55]:
```suggestion
            echo "::error file=$flake_group::${{ env.REPORT }}"
```

Comment (ysndr) [reply .github/workflows/check-flake-files.yml:55]: Tack

**Comment by ysndr** [comment]
I don't know if the explicit exit status is necessary.
Its also a shame that we have to either fail or succeed since this check may fail independently from the flake modified by a PR.
Yet showing success is probably even worse.

**Pull request merged**